src/transformers/models/maskformer/configuration_maskformer.py

```diff
@@ -0,0 +1,150 @@
+}
+
+logger = logging.get_logger(__name__)
+
+RGB = Tuple[int, int, int]
+
+
+@dataclass
+class ClassSpec:
+    is_thing: bool
+    label: str
+    colour: RGB
+
+
+@dataclass
+class DatasetMetadata:
+    classes: List[ClassSpec] = field(default_factory=list)
+
+
+class MaskFormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`MaskFormer`]. It is used to instantiate a MaskFormer
+    model according to the specified arguments, defining the model architecture. Instantiating a configuration with the
+    defaults will yield a similar configuration to that of the "maskformer_swin_base_IN21k_384_bs16_160k_res640" architecture trained on ade20k-150
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+    """
+    model_type = "mask_former"
+
+    attribute_map = {"hidden_size": "d_model"}
+
+    def __init__(
+        self,
+        dataset_metadata: DatasetMetadata = None,
+        fpn_feature_size: Optional[int] = 256,
+        mask_feature_size: Optional[int] = 256,
+        num_classes: Optional[int] = 150,
+        no_object_weight: Optional[float] = 0.1,
+        num_queries: Optional[int] = 100,
+        swin_pretrain_img_size: Optional[int] = 384,
+        swin_in_channels: Optional[int] = 3,
+        swin_patch_size: Optional[int] = 4,
+        swin_embed_dim: Optional[int] = 128,
+        swin_depths: Optional[List[int]] = None,
+        swin_num_heads: Optional[List[int]] = None,
+        swin_window_size: Optional[int] = 12,
+        swin_drop_path_rate: Optional[float] = 0.3,
+        dice_weight: Optional[float] = 1.0,
+        ce_weight: Optional[float] = 1.0,
+        mask_weight: Optional[float] = 20.0,
+        mask_classification: Optional[bool] = True,
+        max_position_embeddings: Optional[int] = 1024,
+        encoder_layers: Optional[int] = 6,
+        encoder_ffn_dim: Optional[int] = 2048,
+        encoder_attention_heads: Optional[int] = 8,
+        decoder_layers: Optional[int] = 6,
+        decoder_ffn_dim: Optional[int] = 2048,
+        decoder_attention_heads: Optional[int] = 8,
+        encoder_layerdrop: Optional[int] = 0.0,
+        decoder_layerdrop: Optional[int] = 0.0,
+        d_model: Optional[int] = 256,
+        dropout: Optional[int] = 0.1,
+        attention_dropout: Optional[int] = 0.0,
+        activation_dropout: Optional[int] = 0.0,
+        init_std: Optional[int] = 0.02,
+        init_xavier_std: Optional[int] = 1.0,
+        scale_embedding: Optional[int] = False,
+        auxiliary_loss: Optional[int] = False,
+        dilation: Optional[int] = False,
+        **kwargs,
+    ):
+        self.dataset_metadata = dataset_metadata
+
+        self.fpn_feature_size = fpn_feature_size
+        self.mask_feature_size = mask_feature_size
+        self.num_classes = num_classes
```

Comment:
```suggestion
```
Each model in Transformers typically use `num_labels` for the number of classes. This is already defined in `PretrainedConfig`, so no need to define it here.